app/models/spree/active_sale_event.rb

```diff
@@ -3,26 +3,22 @@
 # There can be many events/ schedules for one sale.
 #
 module Spree
-  class ActiveSaleEvent < ActiveRecord::Base
-    has_many :sale_images, :as => :viewable, :dependent => :destroy, :order => 'position ASC'
-    has_many :sale_products, :dependent => :destroy, :order => "#{Spree::SaleProduct.table_name}.position ASC"
-    has_many :products, :through => :sale_products, :order => "#{Spree::SaleProduct.table_name}.position ASC"
-    has_many :sale_taxons, :dependent => :destroy, :order => "#{Spree::SaleTaxon.table_name}.position ASC"
-    has_many :taxons, :through => :sale_taxons, :order => "#{Spree::SaleTaxon.table_name}.position ASC"
+  class ActiveSaleEvent < Spree::Base
+    has_many :sale_images, -> { order(position: :asc) }, :as => :viewable, :dependent => :destroy
+    has_many :sale_products, -> { order(position: :asc) }, :dependent => :destroy
+    has_many :products, :through => :sale_products
+    has_many :sale_taxons, -> { order(position: :asc) }, :dependent => :destroy
+    has_many :taxons,  -> { order(position: :asc) }, :through => :sale_taxons
```

Comment:
@ankit1910 - Remove :asc as it is not required. Move order(:position) in a scope.
Use new syntax of Hash.